### DISCUSSION TITLE
Same store for 2 providers

### DISCUSSION THREAD

**Discussion by trish123** [reduxjs/react-redux#2044] [Q&A]
If I give same store for 2 providers, will it be work ? Looks like it works. But not sure If I overlooked some cases? If someone knows,  any help would be appreciated!

Usecase:
 I have some data which I use in Component 1 to render a bar chart and Componet 2 to render a pie chart. New Data can come asynchronously from backend and get stored in redux store. There are actions which can be triggered from the components which affects the data in the store.
Framework loads components. My components are in a Lib. I can ask the Framework  to display Component 1 on the left side in the layout and display Component 2 on the right side in the layout. I need to export Component 1 and Component 2 separately and Framework loads those.

```
const storeInstance= createStore(reducerA);

export const getComponet1 = () => {
return
 <Provider  store = storeInstance>
<Component1/>
</Provider>
}

export const getComponet2 = () => {
return
 <Provider store = storeInstance>
<Component2/>
<Provider>
}

**Comment by markerikson** [comment]
Passing the same store to multiple Providers works fine, yes.

Can you clarify how these components are related, though?  Normally there really should only be one `<Provider>` in the whole component tree, and it should be around the _entire_ component tree (like `<Provider><App /></Provider>`

**Reply by trish123** [reply]
Thanks for the answer @markerikson 

Those 2 components render 2 differnt visualizations from the same data.
Those 2 components are developed by the same lib developer . But the lib developer does NOT own the App. App is a framework which loads these 2 components on differnt part of the UI layout.  For Example, 1st component will be loaded on the left part of the layout and the 2nd component will be loaded on the right part of the layout.

[ please note that I have simplified my answer now]
